core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1803,9 +1805,14 @@ private boolean needNewSubQuery(
 
       if (rel instanceof Aggregate) {
         final Aggregate agg = (Aggregate) rel;
-        final boolean hasNestedAgg = hasNestedAggregations(agg);
+        final boolean hasNestedAgg = hasNested(agg, operand ->
+            operand instanceof SqlCall
```

Comment:
@julianhyde I wasn't sure whether this check should be a part of `hasNested` method or a part of the predicate. I've chosen to go with the more generic predicate approach here, but please let me know your preference.